packages/lodestar-utils/src/json.ts

```diff
@@ -0,0 +1,32 @@
+import {Json, toHexString} from "@chainsafe/ssz";
+
+export function defaultErrorToJson(obj: Error): Json {
+  const errObj: Json = {message: obj.message};
+  if (obj.stack) errObj.stack = obj.stack;
+  return errObj;
+}
+
+interface ICustomError extends Error {
+  toJson?: () => Json;
+}
+
+export function toJson(obj: unknown): Json {
+  if (obj == null) {
+    return null;
+  } else if (typeof obj === "number" || typeof obj === "string") {
+    return obj;
+  } else if (typeof obj === "bigint") {
+    return String(obj);
+  } else if (obj instanceof Uint8Array) {
+    return toHexString(obj);
+  } else if (obj instanceof Error) {
+    return (obj as ICustomError).toJson?.() ?? defaultErrorToJson(obj);
+  } else if (Array.isArray(obj)) {
+    return obj.map(toJson);
+  } else if (typeof obj === "object") {
+    const jsonObj: Json = {};
+    Object.entries(obj as object).forEach(([k, v]) => (jsonObj[k] = toJson(v)));
+    return jsonObj;
+  }
+  throw new Error("Unable to convert unknown to json");
```

Comment:
This throw can potentially prevent the actual error from being logged, `toJson` should never throw and return something

Reply:
This should be unreachable, so it doesn't practically matter. But philosophically, I think it makes sense to error and not "error silently" for a general purpose `toJson` function. See our past struggle with this in the lodestar library. We had many "nullish" functions that simply hide error cases.

Reply:
Then add a stringified json arg to the error message in the edge case it happens we know what data caused the error